### ISSUE TITLE
kustomize cannot set the image without tag

### ISSUE THREAD

**Issue by jinchihe** [kubernetes-sigs/kustomize#1029]
kustomize cannot set the image without tag

From below, I want to set the image (`test_example`) without tag, but printed error, but in the help information, we can see the image can be `- <image>=<newimage>` (without any tag).

```
# kustomize edit set image training-image=test_example
Error: invalid format of image, use one of the following options:
- <image>=<newimage>:<newtag>
- <image>=<newimage>@<newtag>
- <image>=<newimage>
- <image>:<newtag>
- <image>@<digest>
Usage:
  kustomize edit set image [flags]
```
Seems it's a bug?

**Comment by narg95** [comment]
@jinchihe thanks for reporting it, the PR #1063 fixes it

**Comment by jinchihe** [comment]
@narg95 Great, thanks a lot!

**Comment by Liujingfang1** [comment]
Should be fixed by #1063

**Comment by shreya-verma19** [comment]
kustomize edit add configmap mnist-map-training  --from-literal=trainingName=test_kubeflow
Error: cannot add key trainingName, another key by that name already exists: map[GOOGLE_APPLICATION_CREDENTIALS:/var/secrets/user-gcp-sa.json batchSize:100 exportDir:gs://my_bucket_kubeflow/my-model/export learningRate:0.01 modelDir:gs://my_bucket_kubeflow/my-model name:my-train-1 secretMountPath:/var/secrets secretName:user-gcp-sa trainSteps:200 trainingName:test_kubeflow]

But when I am again trying to build it gives error

cloudshell:~/examples/mnist/training/GCS$ kustomize build
Error: no matches for OriginalId kubeflow.org_v1beta2_TFJob|~X|$(trainingName); no matches for CurrentId kubeflow.org_v1beta2_TFJob|~X|$(trainingName); failed to find unique target for patch kubeflow.org_v1beta2_TFJob|$(trainingName)
